test/eval_test.py

```diff
@@ -0,0 +1,434 @@
+import datetime
+
+from eppo_client.models import (
+    Flag,
+    Allocation,
+    Range,
+    VariationType,
+    Variation,
+    Split,
+    Shard,
+)
+from eppo_client.eval import Evaluator, FlagEvaluation, is_in_shard_range, hash_key
+from eppo_client.rules import Condition, OperatorType, Rule
+from eppo_client.sharding import DeterministicSharder, MD5Sharder
+
+VARIATION_A = Variation(key="a", value="A")
+VARIATION_B = Variation(key="b", value="B")
+VARIATION_C = Variation(key="c", value="C")
+
+
+def test_disabled_flag_returns_none_result():
+    flag = Flag(
+        key="disabled_flag",
+        enabled=False,
+        variation_type=VariationType.STRING,
+        variations={"a": VARIATION_A},
+        allocations=[
+            Allocation(
+                key="default", rules=[], splits=[Split(variation_key="a", shards=[])]
+            )
+        ],
+        total_shards=10,
+    )
+
+    evaluator = Evaluator(sharder=MD5Sharder())
+    result = evaluator.evaluate_flag(flag, "subject_key", {})
+    assert result.flag_key == "disabled_flag"
+    assert result.allocation_key is None
+    assert result.variation is None
+    assert not result.do_log
+
+
+def test_matches_shard_full_range():
+    shard = Shard(
+        salt="a",
+        ranges=[Range(start=0, end=100)],
+    )
+
+    evaluator = Evaluator(sharder=MD5Sharder())
+    assert evaluator.matches_shard(shard, "subject_key", 100) is True
+
+
+def test_matches_shard_full_range_split():
+    shard = Shard(
+        salt="a",
+        ranges=[Range(start=0, end=50), Range(start=50, end=100)],
+    )
+
+    evaluator = Evaluator(sharder=MD5Sharder())
+    assert evaluator.matches_shard(shard, "subject_key", 100) is True
```

Comment:
Maybe could add a test with a deterministic sharder to verify that shard `50` is still a match (and it doesn't fall through the cracks).